parser.py

```diff
@@ -121,35 +158,87 @@ def __init__(self, coeff, op):
         self.coeff = coeff
         self.op = op
 
-    def __str__(self):
-        return "{} {}".format(self.coeff, self.op)
-
     def __repr__(self):
-        return self.__str__()
+        return "{} {}".format(self.coeff, self.op)
 
 
 class Declaration:
     def __init__(self, decl_type, **kwargs):
         self.name = decl_type
         self.kwargs = kwargs
+        self.opaque = False
+        self.goplist = []
 
-    def __str__(self):
+    def declaration_str(self):
         return "{} declaration, kwargs={}".format(self.name, self.kwargs)
 
     def __repr__(self):
-        return self.__str__()
+        if self.opaque or not self.goplist:
+            return self.declaration_str()
+
+        output = [self.declaration_str(), "{"]
+
+        for op in self.goplist:
+            output.append(f"\t{op}")
+
+        output.append("}")
+        return "\n".join(output)
+
+
+class QasmProgram:
```

Comment:
At the moment, the QasmProgram is very basic, simply storing a list of statements. We could change this, and have it process the statements to have more structure. Not sure if this is needed for an IR though, I think there should be other functions for processing the IR object.